web/src/app/components/login-page/login-page.component.ts

```diff
@@ -54,32 +54,29 @@ export class LoginPageComponent implements OnInit {
 
   saveUser(user: User, provider: string) {
     this.userService.saveUser(user, provider)
-      .subscribe((userData) => {
-        this.saveUserSocialRecord();
-        this.saveUserSocialStats(userData, provider);
-      });
+      .pipe(
+        map(value => this.saveUserSocialStats(value, provider)),
+        map(() => this.saveUserSocialRecord()),
+        mergeAll()
+      )
+      .subscribe((res) => {
+        this.addProviderToUserSocial(res);
+      }, error => this.errorService.logError(error));
```

Comment:
This can be cleaned up with

```typescript
.subscribe(
   (res) => this.addProviderToUserSocial(res),
   (error) => this.errorService.logError(error)
);
```